### ISSUE TITLE
Analyzer for IAsyncCollector et al.

### ISSUE THREAD

**Issue by mhoeger** [Azure/azure-functions-dotnet-worker#258]
Don't let people use the WebJobs way of output binding

**Comment by onionhammer** [comment]
Better yet, implement support for them

**Comment by kshyju** [comment]
If an `ICollector<T>` or `IAsyncCollector<T>` type is parameter is used **with** any of the WebJobs attribute decoration, our current [WebJobsAttributesNotSupported ](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/sdk/Sdk.Analyzers/WebJobsAttributesNotSupported.cs#L13)analyzer will report an [error](https://dotnet-worker-rules.azurewebsites.net/rules?ruleid=AZFW0001).

<img width="616" alt="image" src="https://user-images.githubusercontent.com/144469/192065396-af29371f-bb89-4c23-ac04-1f3b8cca1052.png">

Same when using out parameter

<img width="604" alt="image" src="https://user-images.githubusercontent.com/144469/192065863-372eea59-7e66-49b7-8b92-515f3d289ff3.png">

But the below one will not report any diagnostic errors.

````
[Microsoft.Azure.WebJobs.StorageAccount("Con")]
public static class Function2
{
    [Function("Function2")]
    [return: Queue("dest-q")]
    public static string Run([QueueTrigger("input-q")] string data) => "foo";
}
````

I feel that, with the existing analyzer, more code samples & docs, we should be good here.  We can keep an eye on the reports from users and based on the data we can revisit this item.

**Comment by fabiocav** [comment]
Let's use this issue to track an enhancement to the existing analyzer so that it handles the return case.

It would also be good to ensure `ICollector` and `IAsyncCollector` cases are handled when using worker extensions.

**Comment by fabiocav** [comment]
Using this to track the fix @kshyju has in place with #1092 

For the type validation, we'll use #1036 to track that work as it covers a broader set of scenarios.

**Comment by fabiocav** [comment]
This work is completed and released with https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker.Sdk/1.8.0-preview2
